tv_apps/browser/js/datasync/sync_ds_helper.js

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+/* global
+  localStorage
+*/
+
+var SyncDsHelper = function (dataStoreName) {
+  var store;
+
+  if (typeof dataStoreName !== 'string') {
+    throw new Event('Incorrect DataStore name.');
+  }
+
+  var init = () => {
+    return _ensureStore();
+  };
+
+  var _ensureStore = () => {
+    if (store) {
+      return Promise.resolve(store);
+    }
+    return navigator.getDataStores(dataStoreName).then(stores => {
+      store = stores[0];
+      return store;
+    });
+  };
+
+  var setSyncedRevId = (revId) => {
+    localStorage.setItem(dataStoreName + '::revId', revId);
```

Comment:
Normally I think this line is OK. But in this case syncing operation could take more than a thousand times. Can we cache the string in advance instead of concatenating them every time? For example ```localStorage.setItem(dataStoreRevId);``` and define ```var dataStoreRevId = '';``` at line 13?